poly-synth/src/poly-synth-demo.js

```diff
@@ -224,6 +248,108 @@ class PolySynthDemo {
+          id: 'playbackRate',
+          type: 'range',
+          min: 0.1,
+          max: 30,
+          step: 0.1,
+          default: this.polySynth_.playbackRate
+        });
+
+    this.drumSampleVolume_ = new ParamController(
+        sideChainDivId,
+        this.polySynth_.setDrumSampleVolume.bind(this.polySynth_), {
+          name: 'Beat volume',
+          id: 'beatVolume',
+          type: 'range',
+          min: 0,
+          max: 5,
+          step: 0.05,
+          default: this.polySynth_.drumVolume
+        });
+
+    this.sideChainThresholdSlider_ = new ParamController(
+        sideChainDivId,
+        this.polySynth_.setNoisegateThreshold.bind(this.polySynth_), {
+          name: 'Threshold',
+          id: 'threshold',
+          type: 'range',
+          min: -100,
+          max: 0,
+          step: 1,
+          default: this.polySynth_.noisegateThreshold
+        });
+
+    this.sideChainAttackSlider_ = new ParamController(
+        sideChainDivId,
+        this.polySynth_.setNoisegateAttack.bind(this.polySynth_), {
+          name: 'Attack',
+          id: 'noisegateAttack',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.05,
+          default: this.polySynth_.noisegateAttack
+        });
+
+    this.sideChainReleaseSlider_ = new ParamController(
+        sideChainDivId,
+        this.polySynth_.setNoisegateRelease.bind(this.polySynth_), {
+          name: 'Release',
+          id: 'noisegateRelease',
+          type: 'range',
+          min: 0,
+          max: 1,
+          step: 0.05,
+          default: this.polySynth_.noisegateRelease
+        });
+
+    document.getElementById('sideChainStartButton').onclick = (event) => {
+      // The change is scheduled slightly into the future to avoid glitching.
+      if (event.target.textContent === 'Start') {
+        event.target.textContent = 'Stop';
+        this.polySynth_.setDrumSample(
+            this.drumSampleBuffers_[this.drumSampleIndex_]);
+        this.polySynth_.activeNoisegateRoute.gain.value = 1;
+        this.polySynth_.bypassNoisegateRoute.gain.value = 0;
+        document.getElementById(this.drumSelectorId_).disabled = true;
+      } else {
+        event.target.textContent = 'Start';
+        this.polySynth_.stopDrumSample();
+        this.polySynth_.activeNoisegateRoute.gain.value = 0;
+        this.polySynth_.bypassNoisegateRoute.gain.value = 1;
+        document.getElementById(this.drumSelectorId_).disabled = false;
+      }
+    }
+
+    let selector = document.getElementById(this.drumSelectorId_);
+    
+    for (let index in this.drumSampleUrls_) {
+      // Only the last part of the file name is displayed.
+      let urlParts = this.drumSampleUrls_[index].split('/');
+      let abbreviatedUrl = urlParts[urlParts.length - 1];
+      let option = document.createElement('option');
+      option.value = index;
+      option.textContent = abbreviatedUrl;
+      selector.appendChild(option);
+    }
+
```

Comment:
You do exactly the same thing for the reverb.  It would be nice to factor this common code out.